zcash_history/src/tree.rs

```diff
@@ -13,21 +13,21 @@ use crate::{Entry, EntryKind, EntryLink, Error, NodeData};
 /// Intended use of this `Tree` is to instantiate it based on partially loaded data (see example
 /// how to pick right nodes from the array representation of MMR Tree), perform several operations
 /// (append-s/delete-s) and then drop it.
-pub struct Tree {
-    stored: HashMap<u32, Entry>,
+pub struct Tree<V: Version> {
+    stored: HashMap<u32, Entry<V>>,
 
     // This can grow indefinitely if `Tree` is misused as a self-contained data structure
-    generated: Vec<Entry>,
+    generated: Vec<Entry<V>>,
```

Comment:
This change looks good for Zebra, we'd just add the same type parameter to our `Tree`, and pass it to the inner `zcash_history::Tree`.